src/rgw/rgw_file.h

```diff
@@ -2322,39 +2322,41 @@ class RGWStatLeafRequest : public RGWLibRequest,
   }
 
   void send_response() override {
-    struct req_state* s = get_state();
+    struct req_state* state = get_state();
     // try objects
     for (const auto& iter : objs) {
       auto& name = iter.key.name;
       lsubdout(cct, rgw, 15) << "RGWStatLeafRequest "
 			     << __func__ << " "
-			     << "list uri=" << s->relative_uri << " "
+			     << "list uri=" << state->relative_uri << " "
 			     << " prefix=" << prefix << " "
 			     << " obj path=" << name << ""
 			     << " target = " << path << ""
 			     << dendl;
       /* XXX is there a missing match-dir case (trailing '/')? */
       matched = true;
-      if (name == path)
+      if (name == path) {
 	exact_matched = true;
-      return;
+        return;
+      }
     }
     // try prefixes
     for (auto& iter : common_prefixes) {
       auto& name = iter.first;
       lsubdout(cct, rgw, 15) << "RGWStatLeafRequest "
 			     << __func__ << " "
-			     << "list uri=" << s->relative_uri << " "
+			     << "list uri=" << state->relative_uri << " "
 			     << " prefix=" << prefix << " "
 			     << " pref path=" << name << " (not chomped)"
 			     << " target = " << path << ""
 			     << dendl;
       matched = true;
       /* match-dir case (trailing '/') */
-      if (name == prefix + "/")
+      if (name == prefix + "/") {
```

Comment:
@mattbenjamin seems like this fix have broken the folllowing s3 test:
```
020-10-05T12:18:54.519 INFO:teuthology.orchestra.run.smithi119.stderr:======================================================================
2020-10-05T12:18:54.519 INFO:teuthology.orchestra.run.smithi119.stderr:FAIL: s3tests_boto3.functional.test_s3.test_bucket_listv2_delimiter_basic
2020-10-05T12:18:54.519 INFO:teuthology.orchestra.run.smithi119.stderr:----------------------------------------------------------------------
2020-10-05T12:18:54.520 INFO:teuthology.orchestra.run.smithi119.stderr:Traceback (most recent call last):
2020-10-05T12:18:54.520 INFO:teuthology.orchestra.run.smithi119.stderr:  File "/home/ubuntu/cephtest/s3-tests/virtualenv/lib/python3.6/site-packages/nose/case.py", line 198, in runTest
2020-10-05T12:18:54.520 INFO:teuthology.orchestra.run.smithi119.stderr:    self.test(*self.arg)
2020-10-05T12:18:54.520 INFO:teuthology.orchestra.run.smithi119.stderr:  File "/home/ubuntu/cephtest/s3-tests/s3tests_boto3/functional/test_s3.py", line 232, in test_bucket_listv2_delimiter_basic
2020-10-05T12:18:54.520 INFO:teuthology.orchestra.run.smithi119.stderr:    eq(response['KeyCount'], len(prefixes) + len(keys))
2020-10-05T12:18:54.520 INFO:teuthology.orchestra.run.smithi119.stderr:AssertionError: 1 != 3
```
either the fix is incorrect or the test has a bug?

Reply:
this change in rgw_file.{h,cc} couldn't change the expected results of that s3-test, I don't think, I can't explain that

Reply:
you are right. it is failing on master as well (when I run s3test locally).